insights/specs/datasources/system_user_dirs.py

```diff
@@ -0,0 +1,75 @@
+"""
+Custom datasource for checking non-root owned packaged directories
+
+Author: Florian Festi <ffesti@redhat.com>
+"""
+import grp
+import json
+import pwd
+import stat
+
+from insights import datasource
+from insights.core.dr import SkipComponent
+from insights.core.spec_factory import DatasourceProvider
+
+try:
+    import rpm
+except ImportError:
+    raise SkipComponent
```

Comment:
@jobselko you'll have to move this import and try/catch down into your datasource.  Its being raised here prior to the core infrastructure invoking the datasource so its not being caught in the same way it would be if raised from the datasource.